src/lib/Libwin/passwd.c

```diff
@@ -4922,7 +4922,7 @@ logon_pw(char *username,
 		strcpy(domain, ".");
 		if (p=strrchr(realname, '\\')) {
 			*p = '\0';
-			strcpy(domain, realname);
+			strncpy(domain, realname, sizeof(domain)/sizeof(domain[0]) - 1);
```

Comment:
We're not using wide characters here, so this can be simplified. We should ensure the string is terminated.
```
strncpy(domain, realname, sizeof(domain) - 1);
domain[sizeof(domain) - 1] = '\0';
```